test/util/test_parse.py

```diff
@@ -40,6 +43,52 @@ def test_extractnumber(self):
+        testExtract("Begin the invasion at 3:45 pm on Thursday", "2017-06-29 15:45:00", "begin invasion")  # noqa
+        testExtract("On Monday, order groceries from Price Chopper", "2017-07-03 00:00:00", "order groceries from price chopper")  # noqa
+        testExtract("Play Happy Birthday music 5 years from today", "2022-06-27 00:00:00", "play happy birthday music")  # noqa
+        testExtract("Skype Mom at 12:45 pm next Thursday", "2017-07-06 12:45:00", "skype mom")  # noqa
+        testExtract("What's the weather next Thursday?", "2017-07-06 00:00:00", "what weather")  # noqa
+        testExtract("what is the weather next friday morning", "2017-07-07 08:00:00", "what is weather")  # noqa
+        testExtract("what is the weather next friday evening", "2017-07-07 19:00:00", "what is weather")  # noqa
+        testExtract("what is the weather next friday afternoon", "2017-07-07 15:00:00", "what is weather")  # noqa
+        testExtract("remind me to call mom on august 3rd", "2017-08-03 00:00:00", "remind me to call mom")  # noqa
+        testExtract("Buy fireworks on the 4th of July", "2017-07-04 00:00:00", "buy fireworks")  # noqa
+        testExtract("what is the weather 2 weeks from next friday", "2017-07-21 00:00:00", "what is weather")  # noqa
+        testExtract("what is the weather wednesday at 0700 hours", "2017-06-28 07:00:00", "what is weather")  # noqa
+        testExtract("what is the weather wednesday at 7 o'clock", "2017-06-28 07:00:00", "what is weather")  # noqa
+        testExtract("Set up an appointment at 12:45 pm next Thursday", "2017-07-06 12:45:00", "set up appointment")  # noqa
+        testExtract("What's the weather this Thursday?", "2017-06-29 00:00:00", "what weather")  # noqa
+        testExtract("set up the appointment for 2 weeks and 6 days from Saturday", "2017-07-21 00:00:00", "set up appointment")  # noqa
+        testExtract("Begin the invasion at 03 45 on Thursday", "2017-06-29 03:45:00", "begin invasion")  # noqa
+        testExtract("Begin the invasion at o 800 hours on Thursday", "2017-06-29 08:00:00", "begin invasion")  # noqa
+        testExtract("Begin the invasion at 8 o'clock in the evening on Thursday", "2017-06-29 20:00:00", "begin invasion")  # noqa
+        testExtract("Begin the invasion at 8 in the evening on Thursday", "2017-06-29 20:00:00", "begin invasion")  # noqa
+        testExtract("Begin the invasion on Thursday at noon", "2017-06-29 12:00:00", "begin invasion")  # noqa
+        testExtract("Begin the invasion on Thursday at midnight", "2017-06-29 00:00:00", "begin invasion")  # noqa
+        testExtract("Begin the invasion on Thursday at 0500", "2017-06-29 05:00:00", "begin invasion")  # noqa
+        testExtract("remind me to wake up in 4 years", "2021-06-27 00:00:00", "remind me to wake up")  # noqa
+        testExtract("remind me to wake up in 4 years and 4 days", "2021-07-01 00:00:00", "remind me to wake up")  # noqa
```

Comment:
Please break these up in two lines each so `# noqa` isn't required. Readability won't be hurt by

```python
        testExtract("Begin the invasion at 8 in the evening on Thursday",
                    "2017-06-29 20:00:00", "begin invasion")
```